services/diddoc/queries/resources/resource_query_collection_id_handler.go

```diff
@@ -0,0 +1,36 @@
+package resources
+
+import (
+	"github.com/cheqd/did-resolver/services"
+	"github.com/cheqd/did-resolver/services/diddoc/queries"
+	"github.com/cheqd/did-resolver/types"
+)
+
+type ResourceCollectionIdHandler struct {
+	queries.BaseQueryHandler
+	ResourceHelperHandler
+}
+
+func (d *ResourceCollectionIdHandler) Handle(c services.ResolverContext, service services.RequestServiceI, response types.ResolutionResultI) (types.ResolutionResultI, error) {
+	resourceCollectionId := service.GetQueryParam(types.ResourceCollectionId)
+	if resourceCollectionId == "" {
+		return d.Continue(c, service, response)
+	}
+
+	// Cast to just list of resources
+	resourceCollection, err := d.CastToContent(service, response)
+	if err != nil {
+		return nil, err
+	}
+
+	// Filter the list of metadatas by the resourceCollectionId
```

Comment:
The noun `metadata` is uncountable. The plural form of metadata is also metadata.